### ISSUE TITLE
[Android] Build failed compileDebugKotlin; Expo 49 & Router v2

### ISSUE THREAD

**Issue by rnnyrk** [expo/expo#23590]
### Summary

I'm trying to create an example with Expo Router V2 and Social Authentication with Supabase for everyone to use. Although I'm struggling a lot with building my project.

Building on iOS works without any issues, but it seems like I can't build the project for Android. 

After running `expo prebuild && expo run:android` 

I keep running into the error:

`Execution failed for task ':expo-modules-core:compileDebugKotlin'.`

<img width="1668" alt="Screenshot 2023-07-18 at 14 36 42" src="https://github.com/expo/expo/assets/4519142/b7b7f22b-83a1-4686-a0fb-f14855794f5c">

I've been trying everything I can find on the internet to make it work.
- pinning Kotlin versions in `build.gradle`
- [Changing JDK versions](https://codereviewvideos.com/how-i-fixed-unsupported-class-file-major-version-64/)
- Adding `android.defaults.buildfeatures.buildconfig=true` in `gradle.properties`, because without that I run into `defaultConfig contains custom BuildConfig fields, but the feature is disabled.` error constantly.
- All the changes to fix to Android build [are in this commit](https://github.com/rnnyrk/expo-router-supabase-social-auth/commit/91d7e3f1fca2dba730857e6d83f6e487839513aa)
- Removing `react-native-reanimated` [as suggested over here](https://github.com/software-mansion/react-native-reanimated/issues/2650)

Seems like there is very little to find about the issues I'm encountering. So any help would be appreciated a lot. 

### Managed or bare workflow?

bare

### What platform(s) does this occur on?

Android

### Package versions

```json
    "app-icon-badge": "0.0.12",
    "dotenv": "16.3.1",
    "expo": "^49.0.3",
    "expo-application": "~5.3.0",
    "expo-build-properties": "~0.8.3",
    "expo-constants": "~14.4.2",
    "expo-dev-client": "~2.4.6",
    "expo-linking": "~5.0.2",
    "expo-router": "2.0.0",
    "expo-secure-store": "~12.3.1",
    "expo-splash-screen": "~0.20.4",
    "expo-status-bar": "~1.6.0",
    "expo-system-ui": "2.4.0",
    "expo-web-browser": "~12.3.2",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-reanimated": "~3.3.0",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-svg": "13.9.0",
    "react-native-url-polyfill": "1.3.0",
    "typescript": "5.1.6"
```

### Environment

```

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.4.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
      npm: 9.5.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
      Watchman: 2023.07.03.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
      Android SDK:
        API Levels: 32, 33, 34
        Build Tools: 30.0.3, 34.0.0
        System Images: android-34 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: ^49.0.3 => 49.0.3
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.72.3 => 0.72.3
    npmGlobalPackages:
      eas-cli: 3.15.1
      expo-cli: 6.3.10
    Expo Workflow: bare
```

### Reproducible demo

The code over here is as minimum as possible for my use case https://github.com/rnnyrk/expo-router-supabase-social-auth.

### EDIT

I just found out that building the project through Android Studio works. Although this messes with my environment settings. So help on fixing the `npm run android` and e.g. `npm run android:production` commands are more than welcome.

**Comment by qalqi** [comment]
+1

**Comment by shawarmaz** [comment]
encountering this problem as well
removing java versions like that just sounds dangerous
surely there is a more precise solution ?

**Comment by stefan-schweiger** [comment]
@shawarmaz as far as I can see the problem is solved with `expo-modules-core@1.5.9`. You can either use overrides in your package.json or force expo to version "^49.0.7".
